client/landing/gutenboarding/components/domain-picker/button.tsx

```diff
@@ -65,28 +67,29 @@ const DomainPickerButton: FunctionComponent = () => {
 	const [ freeDomainSuggestions, paidDomainSuggestions ] = partition( suggestions, 'is_free' );
 
 	return (
-		<Button
-			className="domain-picker__button"
-			onClick={ () => setDomainPopoverVisibility( s => ! s ) }
-		>
-			<div className="domain-picker__site-title">
-				{ siteTitle ? siteTitle : NO__( 'Create your site' ) }
-			</div>
-			<div>{ head( freeDomainSuggestions )?.domain_name }</div>
+		<>
+			<Button
+				aria-expanded={ isDomainPopoverVisible }
+				aria-haspopup="menu"
+				aria-pressed={ isDomainPopoverVisible }
+				className={ classnames( 'domain-picker__button', { 'is-open': isDomainPopoverVisible } ) }
+				onClick={ () => setDomainPopoverVisibility( s => ! s ) }
+			>
+				{ /* This empty span gives us a good place to anchor our popover */ }
+				<span ref={ popoverAnchor } />
+				{ head( freeDomainSuggestions )?.domain_name ?? '\u00a0' }
+				<Dashicon icon="arrow-down-alt2" />
+			</Button>
 			{ isDomainPopoverVisible && (
-				<Popover
-					/* Prevent interaction in the domain picker from affecting the popover */
-					onClick={ e => e.stopPropagation() }
-					onKeyDown={ e => e.stopPropagation() }
-				>
+				<Popover anchorRect={ popoverAnchor.current?.getBoundingClientRect() }>
```

Comment:
I don't see an `anchorRect` prop [here](https://github.com/WordPress/gutenberg/blob/9e74a5f094a55bbce158bea7e47a87dfb18d6d73/packages/components/src/popover/index.js), only a `getAnchorRect` one :thinking:

Reply:
Good spot. Bad types 😱 

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/40524

Fixed in this PR.

Reply:
😕 `anchorRect` is a valid, accepted prop:

https://github.com/WordPress/gutenberg/blob/5a07721207e8b9587031d3d094eca9dd77747c35/packages/components/src/popover/index.js#L262

[This sha](https://github.com/WordPress/gutenberg/blob/9e74a5f094a55bbce158bea7e47a87dfb18d6d73/packages/components/src/popover/index.js) seems to be quite stale.

Reply:
Argh, my bad. I swear I pressed `y` on that file :man_facepalming: